table/format.cc

```diff
@@ -159,6 +160,28 @@ Status ReadBlock(RandomAccessFile* file,
               break;
           }
 
+          case kLZ4Compression: {
+              size_t ulength = DecodeFixed32(data);
+              size_t ret_val;
+              ubuf = new char[ulength];
+
+              ret_val=LZ4_decompress_safe(data+4, ubuf, n-4, ulength);
+              if (ret_val != ulength)
+              {
+                  s=Status::Corruption("corrupted LZ4 compressed block");
+              }   // if
+
+              if (s.ok())
+              {
+                  delete[] buf;
```

Comment:
It would be a good idea to start using smart pointer wrapper classes, when possible, rather than ensuring that explicit deletes are included in all the right places. If std::unique_ptr is available on all platforms where Riak builds, that is the best approach to managing the lifetimes of resources allocated on the heap via new and new[]. If unique_ptr is not available, then you could use std::auto_ptr for single objects and write a similar helper class that deletes arrays of objects.